### PR TITLE
Add Phenom People ATS fetcher

### LINKED ISSUE #4: Add Phenom People fetcher
## Phenom People (Radancy) ATS Support

Phenom is the most common unsupported ATS we've encountered — 4 companies blocked on it.

### API Pattern

Phenom uses a content API at `content-{region}.phenompeople.com` with company-specific `refNum` codes. The careers site is a JS SPA that calls this API.

**Config endpoint (returns widget/site configuration):**
```
GET https://content-{region}.phenompeople.com/api/ddo/getWidgetVersions?refNum={REF}&locale=en_global&siteType=external&deviceType=desktop
```

**Search endpoint (returns job listings):**
```
GET https://content-{region}.phenompeople.com/api/{REF}/globalSearchConfig?locale=en_global&siteType=external&deviceType=desktop
```

Headers required: `Origin` and `Referer` from the company careers domain.

### Companies Waiting

| Company | Careers URL | Region | refNum | Notes |
|---------|------------|--------|--------|-------|
| **RTX / Raytheon** | `careers.rtx.com` | `ir` | `RAYTGLOBAL` | Defense conglomerate, includes Collins Aerospace |
| **BAE Systems** | `jobs.baesystems.com` | `us` | `BAE1US` | Defense, US operations |
| **GE Aerospace** | careers page uses Phenom | — | — | Jet engines, standalone post-GE split |
| **Collins Aerospace** | `careers.rtx.com/global/en/collins-aerospace` | — | — | RTX subsidiary, shares parent Phenom instance |

### Sample curl (BAE Systems)

```bash
curl 'https://content-us.phenompeople.com/api/ddo/getWidgetVersions?refNum=BAE1US&locale=en_global&siteType=external&deviceType=desktop' \
  -H 'origin: https://jobs.baesystems.com' \
  -H 'referer: https://jobs.baesystems.com/'
```

### Sample curl (RTX)

```bash
curl 'https://content-ir.phenompeople.com/api/RAYTGLOBAL/globalSearchConfig?locale=en_global&siteType=external&deviceType=desktop' \
  -H 'origin: https://careers.rtx.com' \
  -H 'referer: https://careers.rtx.com/'
```

### PR DESCRIPTION
**Pull request by coryking**
## Summary

- New stub fetcher for Phenom People ATS (`src/jobbuddy/fetchers/phenom.py`)
- Scrapes career sites via `POST /widgets` API — `refineSearch` DDO for listings, `jobDetail` DDO for full descriptions
- Threaded pagination (5 workers) for large boards
- URL parser with registry reverse-lookup (same pattern as Eightfold)
- Registered in fetcher factory

Covers 4 companies: BAE Systems, RTX/Raytheon, GE Aerospace, Collins Aerospace

## What's in the change

| File | What |
|------|------|
| `src/jobbuddy/fetchers/phenom.py` | PhenomFetcher — stub fetcher with `list_jobs()`, `fetch_job()`, `fetch_description()` |
| `src/jobbuddy/fetchers/__init__.py` | Register phenom in fetcher factory |
| `src/jobbuddy/url.py` | `_parse_phenom()` URL parser |
| `tests/test_phenom.py` | 20 tests covering listing, detail, config, URL parsing |

## Design decisions

- **Stub fetcher** (`descriptions_in_listing = False`) — listing API only returns teasers, not full descriptions. Enrichment phase fetches descriptions via jobDetail DDO.
- **careers_url as config** — each Phenom company needs its own domain (e.g. `jobs.baesystems.com`, `careers.rtx.com`). Passed via Company model_extra, same pattern as Workday's `wd_company`/`wd_instance`.
- **Widget API bypasses Cloudflare** — the JSON API works even when the HTML page is CF-protected (verified on RTX).

## Test plan

- [ ] Verify tests pass with DB available (`uv run python -m pytest tests/test_phenom.py -v`)
- [ ] Register a Phenom company and run `jsb sync fetch --company <name>` to verify real API integration
- [ ] Verify enrichment works: `jsb sync enrich --company <name>`

Closes #4

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by coryking — COMMENTED** [review]
## Review: Phenom People ATS Fetcher

Clean implementation that follows established patterns (Eightfold is the closest analog). Enrichment integration, URL parsing, and test coverage are all solid. A few things to address:

### Must fix

**1. Missing retry logic in `list_jobs()`** — both page 0 and paginated requests call `self.client.post()` raw, without `_retry_request()`. Compare with Eightfold (`eightfold.py:102-124`) which wraps every HTTP call in `_retry_request()`:

```python
# Eightfold page 0 — with retry
data = self._retry_request(_fetch_page0, on_retry=on_retry)

# Eightfold pagination — with retry
def _fetch_page(start: int) -> list[Job]:
    def _do_fetch():
        page_resp = self.client.get(...)
        page_resp.raise_for_status()
        return [...]
    return self._retry_request(_do_fetch, on_retry=on_retry)
```

Phenom has neither. A transient 429 or timeout during listing will crash the entire sync. The `on_retry` callback parameter is accepted but never used — dead code.

**2. Adaptive page size is risky** — `page_size = len(raw_jobs) or PAGE_SIZE` uses the actual first-page response size for offset calculation. If the API returns fewer items than `PAGE_SIZE` on page 0 (filtering quirk, partial response), all subsequent offsets will be wrong and you'll silently skip jobs between pages. Eightfold uses the fixed `PAGE_SIZE` constant for offset calculation. The comment on that line ("the API may return fewer items than the requested `size`") is actually an argument FOR using the constant.

### Nit

**3. No companies registered** — PR description says "Covers 4 companies: BAE Systems, RTX/Raytheon, GE Aerospace, Collins Aerospace" but no company registrations are included. Tests mock them. Might be worth noting in the PR that company registration is a follow-up step via `jsb companies-add`.

### What looks good

- Stub fetcher pattern is correct (`descriptions_in_listing = False`, overrides `fetch_description()`)
- Enrichment integration works: base class `fetch_descriptions()` wraps each `fetch_description()` call with `_retry_request()`, so enrichment gets retry for free ✅
- URL parser follows Eightfold's registry reverse-lookup pattern exactly
- `_require_config()` guard prevents misconfigured fetcher from making requests
- Test coverage is thorough — 20 tests covering listing, detail, config validation, URL parsing, edge cases
- HTML stripping via `strip_html()` applied correctly in both `fetch_job()` and `fetch_description()`

File: src/jobbuddy/fetchers/phenom.py [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+            url=self._job_url(job_id),
+            apply_url=raw.get("applyUrl", self._job_url(job_id)),
+            published_at=_parse_posted_date(raw.get("postedDate")),
+            department=raw.get("category"),
+        )
+
+    def list_jobs(
+        self,
+        *,
+        on_progress: ProgressCallback | None = None,
```
Comment (coryking) [inline src/jobbuddy/fetchers/phenom.py:122]: Missing `_retry_request()` — Eightfold wraps page 0 in `self._retry_request(_fetch_page0, on_retry=on_retry)` ([eightfold.py:107](https://github.com/coryking/jobsearch-buddy/blob/main/src/jobbuddy/fetchers/eightfold.py#L107)). Without this, a transient 429 or timeout here crashes `list_jobs()` entirely.

File: src/jobbuddy/fetchers/phenom.py [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+        refine = data.get("refineSearch", {})
+        total = refine.get("totalHits", 0)
+        raw_jobs = refine.get("data", {}).get("jobs", [])
+        jobs: JobList = [self._parse_listing(j) for j in raw_jobs]
+
+        if on_progress:
+            on_progress(len(jobs), total)
+
+        # Use actual page size from first response for offset calculation —
+        # the API may return fewer items than the requested `size`.
```
Comment (coryking) [inline src/jobbuddy/fetchers/phenom.py]: Same issue in the pagination worker — Eightfold wraps each page fetch in `_retry_request()`:
```python
def _fetch_page(start: int) -> list[Job]:
    def _do_fetch():
        page_resp = self.client.get(...)
        page_resp.raise_for_status()
        return [...]
    return self._retry_request(_do_fetch, on_retry=on_retry)
```
This inner function also needs `_retry_request()`. Also note `on_retry` is accepted as a parameter but never used anywhere — dead code until retry is added.

File: src/jobbuddy/fetchers/phenom.py [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    ) -> JobList:
+        self._require_config()
+        url = self._widgets_url()
+
+        # Fetch page 0 to learn total
+        resp = self.client.post(url, json=self._refine_search_body(0))
+        resp.raise_for_status()
+        data = resp.json()
+
+        refine = data.get("refineSearch", {})
```
Comment (coryking) [inline src/jobbuddy/fetchers/phenom.py:135]: This adaptive page size is fragile. If the API returns fewer items than `PAGE_SIZE` on page 0 (filtering edge case, partial response), all subsequent offsets will be wrong — you'll request overlapping ranges and silently miss jobs in the gaps.

Eightfold uses the fixed `PAGE_SIZE` constant: `remaining_starts = list(range(PAGE_SIZE, count, PAGE_SIZE))`. The comment on this line ("the API may return fewer items") is actually the reason to use the constant, not the response length.
